python/ray/workers/default_worker.py

```diff
@@ -4,6 +4,7 @@
 
 import argparse
 import traceback
+import logging
```

Comment:
alphabetical order of imports (for standard library imports)

Reply:
We run it like this:

```
isort --recursive --check-only --diff ray/
```

And configuration in `setup.cfg` could be something like:

```
[isort]
line_length=1000
known_first_party=ray
indent='    '
comment_prefix='  #'
multi_line_output=2
default_section=THIRDPARTY
combine_as_imports=True
balanced_wrapping=True
include_trailing_comma=True
skip=.svn,CVS,.bzr,.hg,.git,__pycache__,.tox,tests
sections=FUTURE,STDLIB,THIRDPARTY,D3M,FIRSTPARTY,LOCALFOLDER
known_d3m=d3m
```

So it is interesting that you can make its own group. For example, [we use many packages from d3m](https://gitlab.com/datadrivendiscovery), so we have our own section. Ray might want something like arrow maybe in its own section?

Reply:
@mitar How do you handle the conflict between `isort` and `yapf`? `yapf` does not hold a fix style, take the diff output of `yapf` for example:
```
-from ray.ray_constants import (AUTOSCALER_HEARTBEAT_TIMEOUT_S,
-                               AUTOSCALER_MAX_CONCURRENT_LAUNCHES,
-                               AUTOSCALER_MAX_LAUNCH_BATCH,
-                               AUTOSCALER_MAX_NUM_FAILURES,
-                               AUTOSCALER_UPDATE_INTERVAL_S)
+from ray.ray_constants import (
+    AUTOSCALER_HEARTBEAT_TIMEOUT_S, AUTOSCALER_MAX_CONCURRENT_LAUNCHES,
+    AUTOSCALER_MAX_LAUNCH_BATCH, AUTOSCALER_MAX_NUM_FAILURES,
+    AUTOSCALER_UPDATE_INTERVAL_S)
```
and 
```
-from ray.autoscaler.tags import (TAG_RAY_LAUNCH_CONFIG,
-                                 TAG_RAY_NODE_NAME, TAG_RAY_NODE_TYPE)
+from ray.autoscaler.tags import (TAG_RAY_LAUNCH_CONFIG, TAG_RAY_NODE_NAME,
+                                 TAG_RAY_NODE_TYPE)
```
Another thing difference is that `yapf` won't allow empty line between imports and file comments `'''file comments'''`, which `isort` should level 1 empty line. If we set `lines_after_imports=1`, it breaks the rule that there should be 2 empty lines before class or function.

Reply:
We are not using `yapf`, so I don't know. We use pylint and flake8.

Reply:
@guoyuhong When importing multiple items from a module. I prefer to put one item per line:
```
from x.y.z import (
  a,
  b,
  c,
)
```

The reason is because:
1) the item names are naturally sorted.
2) if we add a new item in the future, we only add a new line, without touching  the existing lines. It makes it easy to `git blame` which commit introduced the item. (Note there's a comma in the last line as well.)

Does this style compatible with isort and yapf?

Reply:
@raulchen `Isort` can configure the style but `yapf` cannot do that. `yapf` has a penalty factor, so the style is not fixed. [Style 3](https://github.com/timothycrosley/isort#multi-line-output-modes) is what you want.

Reply:
@guoyuhong I started trying out `pylint` a long time ago, but I think we never ended up using it. We should probably just remove that file.

Reply:
@robertnishihara I gave up trying the `isort` since it has conflict with `yapf`.